sdk/storage/Azure.Storage.Blobs.Cryptography/src/EncryptedBlobClient.cs

```diff
@@ -0,0 +1,323 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlockBlobClient"/>
+        /// class.
+        /// </summary>
+        /// <param name="blobUri">
+        /// A <see cref="Uri"/> referencing the blob that includes the
+        /// name of the account, the name of the container, and the name of
+        /// the blob.
+        /// </param>
+        /// <param name="credential">
+        /// The token credential used to sign requests.
+        /// </param>
+        /// <param name="keyEncryptionKey">
+        /// Required for uploads. Provider to wrap the one-time content encryption key via the envelope technique.
+        /// </param>
+        /// <param name="keyResolver">
+        /// Required for downloads. Provider to get the correct <see cref="IKeyEncryptionKey"/> for a given download.
+        /// The fetched <see cref="IKeyEncryptionKey"/> will be used to unwrap the one-time content encryption key via
+        /// the envelope technique.
+        /// </param>
+        /// <param name="encryptionKeyWrapAlgorithm">
+        /// Required for uploads. String identifier of the key wrapping algorithm the client should use with
+        /// <paramref name="keyEncryptionKey"/>.
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline
+        /// policies for authentication, retries, etc., that are applied to
+        /// every request.
+        /// </param>
+        public EncryptedBlobClient(
+            Uri blobUri,
+            TokenCredential credential,
+            IKeyEncryptionKey keyEncryptionKey = default,
+            IKeyEncryptionKeyResolver keyResolver = default,
+            string encryptionKeyWrapAlgorithm = default,
+            BlobClientOptions options = default)
+            : base(
+                  blobUri,
+                  credential,
+                  options.WithPolicy(new ClientSideDecryptionPolicy(keyResolver, keyEncryptionKey)))
+        {
+            this.KeyWrapper = keyEncryptionKey;
+            this.KeyWrapAlgorithm = encryptionKeyWrapAlgorithm;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlockBlobClient"/>
+        /// class.
+        /// </summary>
+        /// <param name="blobUri">
+        /// A <see cref="Uri"/> referencing the encrypted block blob that includes the
+        /// name of the account, the name of the container, and the name of
+        /// the blob.
+        /// </param>
+        /// <param name="pipeline">
+        /// The transport pipeline used to send every request.
+        /// </param>
+        internal EncryptedBlobClient(Uri blobUri, HttpPipeline pipeline)
+            : base(blobUri, pipeline)
+        {
+            //TODO make new pipeline for this one
+        }
+        #endregion ctors
+
+        /// <summary>
+        /// This method performs a transform on the content stream for uploads. It is a no-op by default.
```

Comment:
This doesn't look like a no-op.

Reply:
Copy/paste from the base method. Will fix.